src/tests/ftest/util/telemetry_test_base.py

```diff
@@ -132,30 +134,30 @@ def get_min_max_mean_stddev(self, suffix, num_targets):
                     "stddev": [0, 1, 2, 3, 4, 5, 6, 7...]
                 }
                 Note that all the stats should have the same rank order for the
-                verification to work. Not using the sample input values.
+                verification to work.
 
         """
         output = {}
 
         for stats_value in ["min", "max", "mean", "stddev"]:
-            specific_metrics = [suffix + stats_value]
+            specific_metrics = [prefix + stats_value]
             pool_out = self.telemetry.get_pool_metrics(
                 specific_metrics=specific_metrics)
             self.log.info("pool_out = %s", pool_out)
 
-            values = [0 for _ in range(num_targets)]
+            values = [0] * total_targets
             offset = 0
 
             for rank_to_dicts in pool_out[specific_metrics[0]].values():
                 for target_to_val in rank_to_dicts.values():
                     for target, value in target_to_val.items():
-                        values[int(target) + offset * 8] = value
+                        values[int(target) + offset * target_per_rank] = value
 
                     offset += 1
```

Comment:
Certainly **_not asking you to change this_**, but just FYI that list append in python is actually fairly efficient compared to pre-allocating a list. Technically slower, but practically negligible for lists smaller than, say, 100K values. So I think it's actually reasonable to do away with `offset` and `target_per_rank` altogether, and do something like:
```
values = []
for ranks in pool_out[specific_metrics[0]].values():
    for targets in ranks.values():
        values += targets.values()
```
Anyway, **_not asking you to change this_**. Coming from a primarily C code base, we tend to do things the C-way :)

Reply:
Thanks for the insight. I pre-allocated so that we're not assuming that the targets are in ascending order. Maybe that's unnecessary assumption and your way is cleaner :)

Reply:
Oh, right. Dictionaries are not sorted by default, so targets.items() might not be in order. Good call ;)